lua/neotest-kotlin/kotest-treesitter-query.lua

```diff
@@ -63,20 +63,7 @@ return [[
     ) 
 ) @test.definition
 
-;; -- todo SHOULD SPEC --
-
-; Matches namespace context("context") { /** body **/ }
-
-(call_expression 
-  (simple_identifier) @function_name (#eq? @function_name "context")
-    (call_suffix 
-      (value_arguments 
-        (value_argument 
-          (string_literal) @namespace.name
-        )
-      ) (annotated_lambda)
-    )
-) @namespace.definition
```

Comment:
This was a bug! Declaring the `context` twice results in multiple nested namespaces that breaks Neotest

Reply:
I'm actually thinking that as an enhancement, I will do a quick scan of the file to determine what kind of spec a given file is and use the appropriate tree-sitter query for the given file. I don't know if this will really have much of a perf impact, but I think it will make these easier to reason about.

Reply:
That approach makes sense, but breaks in more complicated use cases.

A use case I have a lot is with Kotest and Micronaut, where I usually write a top level integration base class to set up Micronaut appropriately that effectively replaces FunSpec, but handles all the test containers set up.

Here's some pseudo code,

```kotlin
abstract class BaseFunSpec(body: BaseFunSpec.() -> Unit) : FunSpec, TestPropertyProvider {
  // This prevents having to nest all your subclass in an init block
  init { body() }

  // Do other test setup
}

// Usage
class Test : BaseFunSpec({
  context("test") {
    test("example") {
      1 shouldBe 1
    }
  }
})
```

Being able to understand what's a FunSpec in this context gets really, really hard without some sort of compilation step though.